synapse/http/federation/matrix_federation_agent.py

```diff
@@ -169,6 +188,37 @@ def _route_matrix_uri(self, parsed_uri):
         service_name = b"_matrix._tcp.%s" % (parsed_uri.host,)
         server_list = yield self._srv_resolver.resolve_service(service_name)
 
+        if not server_list and lookup_well_known:
+            # try a .well-known lookup
+            well_known_server = yield self.get_well_known(parsed_uri.host)
+
+            if well_known_server:
+                # if we found a .well-known, start again, but don't do another
+                # .well-known lookup.
+
+                if b':' in well_known_server:
+                    well_known_host, well_known_port = well_known_server.rsplit(b':', 1)
+                    try:
+                        well_known_port = int(well_known_port)
+                    except ValueError:
+                        well_known_host, well_known_port = well_known_server, -1
```

Comment:
Can you comment what's going on in this section please? Are we saying that `:` is banned in well known response unless its for a port? And if well known lookup fails for any reason we fall back to trying to connect directly?

Separately, should there be a log line here? As I think at this point its a configuration error (rather than someone not setting a well-known), and it'd be nice to log for debuggability.

Reply:
Oh, or is this trying to check if its an IPv6 literal?

Reply:
yes. Added a comment.